### PR TITLE
Add the openldap base module.

### PR DESCRIPTION
**Pull request by leifmadsen**
Add the openldap module after testing.

Signed-off-by: Leif Madsen leif@leifmadsen.com

### CODE REVIEW

**Comment by pabelanger** [comment]
You'll need to rebase and merge the 3 commit messages.

**Comment by leifmadsen** [comment]
Done.

File: openldap/manifests/classes/install.pp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class openldap::classes::install {
+
+	if defined(Class['openldap::server']) {
+		package { "$openldap::params::packagename":
+			name	=> $openldap::params::packagename,
```
Comment (pabelanger) [inline openldap/manifests/classes/install.pp]: name is not needed anymore, it is implied by the first line

File: openldap/manifests/classes/install.pp [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+			ensure	=> present,
+		}
+	}
+
+	package { 'ldap-utils':
```
Comment (pabelanger) [inline openldap/manifests/classes/install.pp]: package { "$openldap::params::packagename_client":

then you can remove the next line

File: openldap/manifests/classes/service.pp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class openldap::classes::service {
+	service { 'slapd':
```
Comment (pabelanger) [inline openldap/manifests/classes/service.pp]: same:

service { "$openldap::params::servicename":

remove next line

File: openldap/files/etc/default/slapd [hunk 4]
```diff
@@ -0,0 +1,47 @@
```
Comment (pabelanger) [inline openldap/files/etc/default/slapd]: I'd rather see these files templates, so we can override some of the hardcoded settings.  EG: SLAPD_USER and SLAPD_GROUP

File: openldap/files/initial_config.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
```
Comment (pabelanger) [inline openldap/files/initial_config.sh]: same comment as above

File: openldap/templates/frontend.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Create top-level object in domain
+dn: <%= ldap_domain %>
```
Comment (pabelanger) [inline openldap/templates/frontend.erb]: Where is this variable coming from?   Perhaps we should consider settings defaults

**Comment by leifmadsen** [comment]
I'm throwing away this pull request. I'll bring another one in that is clean.

**Pull request closed without merging**